Tests/Common/Util/RingBufferTest.swift

```diff
@@ -0,0 +1,59 @@
+@testable import CioInternalCommon
+import Foundation
+import SharedTests
+import XCTest
+
+class RingBufferTest: UnitTest {
+    func test_EnqueueAndDequeue_ShouldHandleElementsInFIFOOrder() {
+        var ringBuffer = RingBuffer<Int>(capacity: 3)
+        ringBuffer.enqueue(1)
+        ringBuffer.enqueue(2)
+
+        XCTAssertEqual(ringBuffer.dequeue(), 1, "Dequeued element should be 1")
+        XCTAssertEqual(ringBuffer.dequeue(), 2, "Dequeued element should be 2")
+        XCTAssertNil(ringBuffer.dequeue(), "Buffer should be empty")
+    }
+
+    func test_EnqueueBeyondCapacity_ShouldOverwriteOldestElement() {
+        var ringBuffer = RingBuffer<Int>(capacity: 2)
+        ringBuffer.enqueue(1)
+        ringBuffer.enqueue(2)
+        ringBuffer.enqueue(3) // This should overwrite the first element (1)
```

Comment:
the comment confused me but the test assertion below matched what I would expect from a queue

Reply:
whats its saying is since the `limit is 2`, the last element when `enqued` should kick the 1st element out, but i understand what you mean.